manifests/service/solaris.pp

```diff
@@ -13,5 +13,18 @@
     service { 'pe-puppet':
       ensure => stopped,
     }
+  } elsif $::operatingsystem == 'Solaris' and $::operatingsystemmajrelease == '11' {
+    file { '/tmp/solaris_start_puppet.sh':
+      ensure  => file,
+      content => template('puppet_agent/solaris_start_puppet.sh.erb'),
+      mode    => '0755',
+    } ->
+    exec { 'solaris_start_puppet.sh':
```

Comment:
Mcollective is handled in the same script: https://github.com/puppetlabs/puppetlabs-puppet_agent/pull/73/files#diff-f2d66952a483ed099ff3363202e04b69R14